codebase/git.go

```diff
@@ -0,0 +1,67 @@
+package codebase
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+)
+
+type Scheme string
+
+const (
+	HTTPS Scheme = "https"
+)
+
+// There is lot with Git URL - supported scheme, format and validation (regex). Currently, GitURL support very specific conversion.
+// reference:
+// https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-clone.html#_git_urls_a_id_urls_a
+// https://github.com/git/git/blob/77bd3ea9f54f1584147b594abc04c26ca516d987/urlmatch.c
+// https://stackoverflow.com/questions/2514859/regular-expression-for-git-repository
+// https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols
```

Comment:
Parsing any_git_url and Converting to different scheme required lot of implementation and efforts.
Currently,
design - provide complete design to support different scheme conversion in future.
implementation - provided basic implementation to support HTTPS conversion which is needed for WIT codebase search.